tools/local-sensor/main.go

```diff
@@ -171,13 +183,30 @@ func mustGetCommandLineArgs() localSensorConfig {
 	return sensorConfig
 }
 
-func registerHostKillSignals(startTime time.Time, fakeCentral *centralDebug.FakeService, outfile string, outputFormat string, cancelFunc context.CancelFunc) {
+func writeMemoryProfile() {
+	f, err := os.Create(fmt.Sprintf("local-sensor-mem-%s.prof", time.Now().UTC().Format(time.RFC3339)))
+	if err != nil {
+		log.Fatal("could not create memory profile: ", err)
+	}
+	defer utils.IgnoreError(f.Close)
+	runtime.GC()
+	if err := pprof.Lookup("allocs").WriteTo(f, 0); err != nil {
+		log.Fatal("could not write memory profile: ", err)
+	}
+	log.Printf("Wrote memory profile")
+}
+
+func registerHostKillSignals(startTime time.Time, fakeCentral *centralDebug.FakeService, writeMemProfile bool, outfile string, outputFormat string, cancelFunc context.CancelFunc) {
 	ctx, stop := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
 	defer stop()
 	<-ctx.Done()
 	// We cancel the creation of Events
 	cancelFunc()
 	endTime := time.Now()
+	if writeMemProfile {
+		writeMemoryProfile()
+	}
+	pprof.StopCPUProfile()
```

Comment:
Shouldn't this be guarded by the the `NoCPUProfile` flag as well? I wonder if `pprof` would panic/err if `StopCPUProfile` is called when `StartCPUProfile` wasn't.

Reply:
I tested all combinations and there was no panic. I needed to place it here, because when I put it in the deferred statement, the CPU profiles were empty. Maybe because we need to kill sensor with ctrl+c? I didn't look deeper